framework/Models.py

```diff
@@ -1468,59 +1451,81 @@ def createNewInput(self,currentInput,samplerType,**Kwargs):
     if not found:
       self.raiseAnError(IOError,'None of the input files has one of the extensions requested by code '
                                   + self.subType +': ' + ' '.join(self.code.getInputExtension()))
-    Kwargs['outfile'] = 'out~'+newInputSet[index].getBase()
-    subDirectory = os.path.join(self.workingDir,Kwargs['prefix'] if 'prefix' in Kwargs.keys() else '1')
+    subDirectory = os.path.join(self.workingDir, kwargs['prefix'] if 'prefix' in kwargs.keys() else '1')
 
     if not os.path.exists(subDirectory):
       os.mkdir(subDirectory)
+
     for index in range(len(newInputSet)):
       newInputSet[index].setPath(subDirectory)
       shutil.copy(self.oriInputFiles[index].getAbsFile(),subDirectory)
-    Kwargs['subDirectory'] = subDirectory
-    if 'SampledVars' in Kwargs.keys():
-      sampledVars = self._replaceVariablesNamesWithAliasSystem(Kwargs['SampledVars'],'input',False)
-    newInput    = self.code.createNewInput(newInputSet,self.oriInputFiles,samplerType,**copy.deepcopy(Kwargs))
-    if 'SampledVars' in Kwargs.keys() and len(self.alias['input'].keys()) != 0:
-      Kwargs['SampledVars'] = sampledVars
-    return (newInput,Kwargs)
-
-  def updateInputFromOutside(self, Input, externalDict):
-    """
-      Method to update an input from outside
-      @ In, Input, list, list of inputs that needs to be updated
-      @ In, externalDict, dict, dictionary of new values that need to be added or updated
-      @ Out, inputOut, list, updated list of inputs
-    """
-    newKwargs = Input[1]
-    newKwargs['SampledVars'].update(externalDict)
-    # the following update should be done with the Pb value coming from the previous (in the model chain) model
-    newKwargs['SampledVarsPb'].update(dict.fromkeys(externalDict.keys(),1.0))
-    self._replaceVariablesNamesWithAliasSystem(newKwargs['SampledVars'  ],'input',False)
-    self._replaceVariablesNamesWithAliasSystem(newKwargs['SampledVarsPb'],'input',False)
-    inputOut = self.createNewInput(Input[1]['originalInput'], Input[1]['SamplerType'], **newKwargs)
-    return inputOut
-
-  def run(self,inputFiles,jobHandler):
-    """
-      Method that performs the actual run of the Code model
-      @ In,  Input, object, object contained the data to process. (inputToInternal output)
-      @ In,  jobHandler, JobHandler instance, the global job handler instance
-      @ Out, None
+
+    kwargs['subDirectory'] = subDirectory
+
+    if 'SampledVars' in kwargs.keys():
+      sampledVars = self._replaceVariablesNamesWithAliasSystem(kwargs['SampledVars'],'input',False)
```

Comment:
Good question. This did come from the existing code and was only displaced by my edits. I am seeing another piece of code right below this that @alfoa had already commented on some place else, so it would appear there is some redundancy in the separate Models class.  Maybe, I should open an issue regarding some repeated use of different snippets of code in this file?